src/plugins/data/server/search/search_service.ts

```diff
@@ -140,7 +140,7 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
       .pipe(first())
       .toPromise()
       .then((value) => {
-        if (value.search.aggs.shardDelay.enabled) {
```

Comment:
is this always undefined? if so it's not an error right otherwise could this conditional be completely removed?

Reply:
I agree with you that if `value.search.aggs.shardDelay.enabled` is always undefined, then removed this completely. But the question is how can we know it is always undefined? 

`src/plugins/data/server/search/search_service.ts` does have `value` tested as null therefore `value.search.aggs` is undefined. I am not sure whether in real application usage, `value` or `value.search` will always be undefined or not, so I just add conditional check for each one. I think that is more logically restrict. What do you think?

Reply:
I think it's fine but almost would prefer to have a TODO here to figure if this is always undefined then merge in your current fix to unblock us.